ci: push cilium-test-dev image to quay, accept tags in the test script

nebril opened the pull request:
We want to have this image on quay because of docker rate limiting

nebril committed:
ci: push cilium-test-dev image to quay

Signed-off-by: Maciej Kwiek <maciej@isovalent.com>

errordeveloper commented on images/Makefile:
```diff
@@ -1,9 +1,7 @@
-REGISTRIES ?= docker.io/cilium
-# quay.io is not enabled, see https://github.com/cilium/image-tools/issues/11
-# REGISTRIES ?= docker.io/cilium quay.io/cilium
+REGISTRIES ?= docker.io/cilium quay.io/cilium
```
I don't think we can enable this for all images just yet, I'm not entirely sure if all images have been updated since https://github.com/cilium/image-tools/pull/59, please feel free to check as part of this.

errordeveloper commented on .github/workflows/images.yaml:
```diff
@@ -75,6 +75,8 @@ jobs:
+          QUAY_USERNAME: ${{ secrets.QUAY_USERNAME  }}
         with:
           entrypoint: make
           args: -C images cilium-test-image PUSH=true
```
You can just add `REGISTRIES="docker.io/cilium quay.io/cilium"` on this line to achieve what you want the title of this PR suggests.

errordeveloper reviewed (changes requested):
See comments

nebril committed:
image: Allow cilium-test-gke.sh to specify tags

Allow script for test-running script to specify tags for images used in
test run

Signed-off-by: Maciej Kwiek <maciej@isovalent.com>
